src/viewers/viewer/Viewer.js

```diff
@@ -380,8 +380,15 @@ class Viewer extends OlObject {
         var dims = this.image_info_['size'];
         if (this.image_info_['zoomLevelScaling']) {
             var tmp = [];
-            for (var r in this.image_info_['zoomLevelScaling'])
-                tmp.push(1 / this.image_info_['zoomLevelScaling'][r]);
+            for (var r in this.image_info_['zoomLevelScaling']) {
+                var scale = 1 / this.image_info_['zoomLevelScaling'][r];
```

Comment:
to avoid surprises, I think we need to add a check that ``this.image_info_['zoomLevelScaling'][r]`` is not ``0``

Reply:
These values come from the server via https://github.com/ome/omero-py/blob/99ff5fd87b327a774dc55613b5d639fce93f4eaf/src/omero/gateway/__init__.py#L8714.
So it is very unlikely we will get a `0` here, and if so then there is something very wrong with the image so we won't be able to provide an alternative value here that will allow the image to be viewed.
And I think it's a bit of an edge case to be implementing an error dialog etc.

Reply:
could you add just a note ``unlikely to get 0`` coming from ...
so we know that is expected